packages/blob-storage-manager/src/storages/FileSystemStorage.ts

```diff
@@ -0,0 +1,80 @@
+import fs from "fs";
+import path from "path";
+
+import { BlobStorage } from "../BlobStorage";
+import type { BlobStorageConfig } from "../BlobStorage";
+import type { Environment } from "../env";
+import { BlobStorageError } from "../errors";
+import { BLOB_STORAGE_NAMES } from "../utils";
+
+export interface FileSystemStorageConfig extends BlobStorageConfig {
+  blobDirPath: string;
+}
+
+export class FileSystemStorage extends BlobStorage {
+  blobDirPath: string;
+
+  constructor({ blobDirPath, chainId }: FileSystemStorageConfig) {
+    super(BLOB_STORAGE_NAMES.FILE_SYSTEM, chainId);
+
+    this.blobDirPath = blobDirPath;
+
+    if (!fs.existsSync(this.blobDirPath)) {
+      fs.mkdirSync(this.blobDirPath);
+    }
+  }
+
+  protected async _healthCheck(): Promise<void> {
+    return Promise.resolve();
```

Comment:
I wonder if we should check if the storage directory exists and the process has write permissions in it?

Reply:
The storage creates the directory automatically if it doesn't exists. See [here](https://github.com/Blobscan/blobscan/blob/f5a8977e2ded92b3aadd8aeb493ac596e3076613/packages/blob-storage-manager/src/storages/FileSystemStorage.ts#L22).

Should we create it manually? I personally prefer for the storage to do it in case it doesn't exists